### ISSUE TITLE
C#/Mono [Export] List<>variable

### ISSUE THREAD

**Issue by brmassa** [godotengine/godot#20387]
Currently using [Export] with a C# List produces nothing on the inspector. Even worse, when it have a initial value (event an empty list), it crashes the editor.

```
Class CustomNode : Node
{
 [Export]
 List<string> list = new List<string>();
}
```

**Comment by brmassa** [comment]
Is there any work around to create lists using C#?

**Comment by akien-mga** [comment]
#35601 includes a reproduction project matching current 3.2 RC 3: https://github.com/godotengine/godot/issues/35601#issue-555289057

**Comment by nathanwfranke** [comment]
As I posted in the issue akien linked, I am getting bizarre behaviour.

Code:  
```
[Export]
public List<int> list {get; set;}

public override void _Ready()
{
	GD.Print("Length = " + list.Count);
	GD.Print("Values = " + string.Join(", ", list));
}
```

![image](https://user-images.githubusercontent.com/14253836/73219017-c8b0a480-4120-11ea-8dcb-b8ac3a5b4b13.png)
![image](https://user-images.githubusercontent.com/14253836/73219028-cd755880-4120-11ea-9f50-271d83499c71.png)

**Comment by DeathTBO** [comment]
I just tested it in Godot 3.2, and oh man, I thought something seemed off. The List is not null, always returns a count of 0, and according to @nathanwfranke shows items when doing string.Join.

It also crashes when attempting to access an item with an index and foreach doesn't do anything at all.

String.Join works though. This uses an enumerator to manually go through the items. So I setup my own. It works perfectly:

```
  IEnumerable<T> IterateList<T>(IEnumerable<T> collection)
	{
		if (collection == null)
			throw new NullReferenceException();

		using (IEnumerator<T> enumerator = collection.GetEnumerator())
		{
			while (enumerator.MoveNext())
			{
				yield return enumerator.Current;
			}
		}
	}
```
I tested it with NodePaths, ints, strings. They all work just fine. Hopefully this gets fixed soon because this workaround feels hacky to me.

**Comment by nobody0** [comment]
Same issue, and even using the "IterateList" method it still crashes as soon as i try to access any property of the Resource(in my case).
If i use a Godot.Collections.Array instead i always get the correct length but the crash when accessing properties persists.

**Comment by neikeq** [comment]
Should have been fixed by #38124
